File: backend/app/schemas/ai_analysis.py
"""Pydantic schemas for AI analyses (CU-32 / CU-33)."""

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AIAnalysisOut(BaseModel):
    # protected_namespaces=(): permite el campo `model_version` sin warning.
    model_config = ConfigDict(from_attributes=True, protected_namespaces=())

    id: UUID
    claim_id: UUID | None
    claim_request_id: UUID | None
    evidence_id: UUID | None
    kind: str
    status: str
    score: Decimal | None
    payload: dict | None
    explanation: str | None
    model_version: str | None
    created_at: datetime
    updated_at: datetime | None


class AIAnalysisListResponse(BaseModel):
    items: list[AIAnalysisOut]


class AIRefreshResponse(BaseModel):
    detail: str
    enqueued: bool

File: frontend/types/ai-analysis.ts
// Ciclo 8 — CU-32: contratos de ai_analyses (espejo de schemas/ai_analysis.py)

export type AIAnalysisKind = 'inconsistency' | 'duplicate' | 'fraud_score' | 'damage_assessment'
export type AIAnalysisStatus = 'processing' | 'done' | 'error'

export interface InconsistencyFinding {
  severity: 'info' | 'warning' | 'critical'
  field: string
  message: string
}

export interface DuplicateMatch {
  claim_id: string
  claim_number: string
  similarity: number
}

export interface FraudFactor {
  name: string
  contribution: number
  direction: 'up' | 'down'
}

export interface DamageAssessment {
  damage_type: string
  severity: 'leve' | 'moderado' | 'severo' | 'indeterminado'
  confidence: number
}

export interface AIAnalysis {
  id: string
  claim_id: string | null
  claim_request_id: string | null
  evidence_id: string | null
  kind: AIAnalysisKind
  status: AIAnalysisStatus
  score: number | null
  // Forma según kind: {findings} | {matches} | {score, factors} | DamageAssessment
  payload: Record<string, unknown> | null
  explanation: string | null
  model_version: string | null
  created_at: string
  updated_at: string | null
}

export interface AIAnalysisListResponse {
  items: AIAnalysis[]
}
